src/subscript/welltest_dpds/welltest_dpds.py

```diff
@@ -0,0 +1,475 @@
+    print("Extracting result from well:", well_name)
+    print()
+
+    if outf_suffix and not outf_suffix.startswith("_"):
+        outf_suffix = "_" + outf_suffix
+
+    if not outdir.endswith("/"):
+        outdir = outdir + "/"
+
+    if not os.path.exists(outdir):
+        raise FileNotFoundError("No such outputdirectory:", outdir)
+
+    summary = EclSum(eclcase)
+    time = np.array(summary.days) * 24.0
+    wbhp = get_summary_vec(summary, "WBHP:" + well_name)
+
+    if main_phase == "OIL":
+        wopr = get_summary_vec(summary, "WOPR:" + well_name)
+        wgpr = get_summary_vec(summary, "WGPR:" + well_name, required=False)
+    else:
+        wopr = get_summary_vec(summary, "WOPR:" + well_name, required=False)
+        wgpr = get_summary_vec(summary, "WGPR:" + well_name)
+
+    wwpr = get_summary_vec(summary, "WWPR:" + well_name, required=False)
+
+    if main_phase == "OIL":
+        buildup_indices, buildup_end_indices = get_buildup_indices(wopr)
+    else:
+        buildup_indices, buildup_end_indices = get_buildup_indices(wgpr)
+
+    print("Identified %d buildup periods:" % (len(buildup_indices)))
+    print("Starting at time steps:" + str(buildup_indices) + " Corresponding to:")
+    for i in buildup_indices:
+        print("  %0.5f Hours" % (time[i]))
+    print("Ending at time step:" + str(buildup_end_indices) + " Corresponding to: ")
+    for i in buildup_end_indices:
+        print("  %0.5f Hours" % (time[i]))
+
+    if buildup_nr > len(buildup_indices):
+        sys.stderr.write(
+            "There are %d build ups detected for well %s. You asked for nr %d\n"
+            % (len(buildup_indices), well_name, buildup_nr)
+        )
+        sys.exit(1)
+    else:
+        bu_start_ind = buildup_indices[buildup_nr - 1]
+        print(
+            "Selected buildup period is nr %d, starting at: %0.5f Hours"
+            % (buildup_nr, time[buildup_indices[buildup_nr - 1]])
+        )
+    print()
+
+    # Find end of buildup period.
+    bu_end_ind = buildup_end_indices[buildup_nr - 1]
+
+    if main_phase == "OIL":
+        super_time = get_supertime(time, wopr, bu_start_ind, bu_end_ind)
+    else:
+        super_time = get_supertime(time, wgpr, bu_start_ind, bu_end_ind)
+
+    # Calculate delta pressure             - lag 1 only
+    dp = np.diff(wbhp[bu_start_ind + 1 : bu_end_ind + 1])
+
+    # Calculate delta superpositioned time - lag 1 only
+    dspt = np.diff(super_time)  # Supertime at Tn is not defined.
+
+    # Cumulative time used from start of buildup
+    cum_time = time[bu_start_ind + 1 : bu_end_ind + 1] - time[bu_start_ind]
+
+    dpdspt_weighted_lag1 = get_weighted_avg_press_time_derivative_lag1(dp, dspt)
+
+    dpdspt_weighted_lag2 = get_weighted_avg_press_time_derivative_lag2(
+        dp, dspt, super_time, wbhp, bu_start_ind, bu_end_ind
+    )
+
+    to_csv(
+        outdir + "dpdspt_lag1" + outf_suffix + ".csv",
+        [cum_time, dpdspt_weighted_lag1],
+        ["Hours", "dpd(supt)_w"],
```

Comment:
I think I would prefer it this was written "HOUR" and not "Hours", for consistency with other tools.